File: Practice(Beginner)/Python3/Id_and_Ship.py
# Source: https://www.codechef.com/problems/FLOW010

n = int(input())
d = {"b":"BattleShip","B":"BattleShip","c":"Cruiser","C":"Cruiser","d":"Destroyer","D":"Destroyer","f":"Frigate","F":"Frigate"}
for i in range(n):
    x = input()
    print(d[x])

File: Practice(Beginner)/C++/Id and Ship.cpp
// Source: https://www.codechef.com/problems/FLOW010

#include <bits/stdc++.h>
using namespace std;

string shipClass(char ch){
    unordered_map <char, string> shipName;
    shipName['b'] = "BattleShip";
    shipName['B'] = "BattleShip";
    shipName['c'] = "Cruiser";
    shipName['C'] = "Cruiser";
    shipName['d'] = "Destroyer";
    shipName['D'] = "Destroyer";
    shipName['f'] = "Frigate";
    shipName['F'] = "Frigate";
    return shipName[ch];
}

int main() {
    int t;
    char ch;
    cin >> t;
    while (t--){
        cin >> ch;
        cout << shipClass(ch) << endl;
    }
    return 0;
}